genie-agent/src/main/java/com/netflix/genie/agent/GenieAgentApplication.java

```diff
@@ -0,0 +1,84 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent;
+
+import com.beust.jcommander.ParameterException;
+import com.netflix.genie.agent.cli.AgentCLICommand;
+import com.netflix.genie.agent.cli.ArgumentParser;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.jmx.JmxAutoConfiguration;
+import org.springframework.boot.autoconfigure.web.WebClientAutoConfiguration;
+
+import java.util.Arrays;
+
+
+/**
+ * Genie Agent application.
+ */
+@SpringBootApplication(
```

Comment:
I think we should start targeting Boot 2.0 now as it will have better support for command line programs and will be our eventual platform. At the very least for the Agent we can pin dependencies to the 2.0 release train if we want to leave Web running on 1.5.x for a bit but I was planning on following the [migration guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide) soon provided @twicksell had at least some internal support for 2.0 so our dev build could still deploy

Reply:
What they mention here: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.0-Migration-Guide#using-the-applicationrunner-or-commandlinerunner is in fact something i was confused about, and I was wondering if i was doing something wrong.
Other than that minor difference, upgrading to 2.0 now doesn't seem to make a difference with the agent.
Can I wait until you upgrade, rather than going through the trouble of pinning (or alternatively, I can start the upgrade project-wide myself)